base/essentials.jl

```diff
@@ -590,6 +590,33 @@ macro goto(name::Symbol)
     return esc(Expr(:symbolicgoto, name))
 end
 
+"""
+    @repeat n call
+    @repeat bool_expr call
+
+Repeat `call` `n` times and discard the output.
+If an expression that returns a boolean is given as a first argument, repeat while true and disregard output.
+"""
+macro repeat(terms, ex)
+    if isa(terms, Integer)
```

Comment:
It seems confusing to me to have a different behavior depending on whether the first argument is a literal integer here. I would expect
```julia
n = 2
@repeat n println("foo")
```
to work differently. Also note that `Bool <: Integer`, so `@repeat true foo()` will only run once and not be an infinite loop.

Reply:
Ah yes. Given the expr could return anything, I've made it a requirement that the first expr returns a Bool and assumed a while loop is wanted.
But it is unfortunately odd UX that the macro can't infer what to do here properly
```julia
julia> @repeat 3 + 1 println("hello")
ERROR: TypeError: non-boolean (Int64) used in boolean context
Stacktrace:
 [1] top-level scope
   @ ./essentials.jl:613
```

I've fixed `@repeat true foo()`, thanks for that pointer, and added `@repeat foo()`